views/contexts/view.go

```diff
@@ -0,0 +1,728 @@
+	if end > len(m.fileBrowserFiles) {
+		end = len(m.fileBrowserFiles)
+		start = end - maxVisible
+		if start < 0 {
+			start = 0
+		}
+	}
+
+	for i := start; i < end; i++ {
+		filePath := m.fileBrowserFiles[i]
+		var displayName string
+		if filePath == ".." {
+			displayName = "📁 .."
+		} else {
+			displayName = filepath.Base(filePath)
+			// Show directory indicator
+			if strings.HasSuffix(filePath, "/") {
+				displayName = "📁 " + strings.TrimSuffix(displayName, "/")
+			}
+		}
+		if i == m.fileBrowserCursor {
+			lines = append(lines, selectedStyle.Render("→ "+displayName))
+		} else {
+			lines = append(lines, itemStyle.Render("  "+displayName))
+		}
+	}
+
+	lines = append(lines, itemStyle.Render(""))
+	helpText := fmt.Sprintf(" %s Select/Navigate • %s / %s Move • %s Cancel",
+		keyStyle.Render("<Enter>"),
+		keyStyle.Render("<↑/↓>"),
+		keyStyle.Render("<PgUp/PgDn>"),
+		keyStyle.Render("<Esc>"))
+	lines = append(lines, helpStyle.Render(helpText))
+
+	content := lipgloss.JoinVertical(lipgloss.Left, lines...)
+	return borderStyle.Render(content)
+}
+
+// renderErrorDialog renders the error dialog
+func (m *Model) renderErrorDialog() string {
+	titleStyle := lipgloss.NewStyle().
+		Bold(true).
+		Foreground(lipgloss.Color("15")).
+		Background(lipgloss.Color("196")). // Red background for error
+		Padding(0, 1)
+
+	borderStyle := lipgloss.NewStyle().
+		Border(lipgloss.RoundedBorder()).
+		BorderForeground(lipgloss.Color("196")) // Red border
+
+	itemStyle := lipgloss.NewStyle().
+		Padding(0, 1)
+
+	helpStyle := lipgloss.NewStyle().
+		Foreground(lipgloss.Color("240")).
+		Padding(0, 1)
+
+	keyStyle := lipgloss.NewStyle().
+		Foreground(lipgloss.Color("63")).
+		Bold(true)
+
+	var lines []string
+	lines = append(lines, titleStyle.Render(" Error "))
+	lines = append(lines, itemStyle.Render(""))
+
+	// Wrap error message if too long
+	errorMsg := m.GetError()
+	maxWidth := 70
+	wrappedLines := wrapText(errorMsg, maxWidth)
+	for _, line := range wrappedLines {
+		lines = append(lines, itemStyle.Render(line))
+	}
+
+	lines = append(lines, itemStyle.Render(""))
+	helpText := fmt.Sprintf("%s %s %s",
+		helpStyle.Render("Press"),
+		keyStyle.Render("<Enter>"),
+		helpStyle.Render("to close"))
+	lines = append(lines, helpText)
+
+	content := lipgloss.JoinVertical(lipgloss.Left, lines...)
+	return borderStyle.Render(content)
+}
+
+// wrapText wraps text to specified width
+func wrapText(text string, width int) []string {
+	if len(text) <= width {
+		return []string{text}
+	}
+
+	var lines []string
+	words := strings.Fields(text)
+	currentLine := ""
+
+	for _, word := range words {
+		if len(currentLine)+len(word)+1 <= width {
+			if currentLine == "" {
+				currentLine = word
+			} else {
+				currentLine += " " + word
+			}
+		} else {
+			if currentLine != "" {
+				lines = append(lines, currentLine)
+			}
+			currentLine = word
+		}
+	}
+
+	if currentLine != "" {
+		lines = append(lines, currentLine)
+	}
+
+	return lines
+}
```

Comment:
All those dialogs are very generic. I suggest moving them to the ui/components package, but you can skip this now. Just please do not duplicate them when you need them somewhere else.